diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index a9969f5..f02fe7e 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -89,13 +89,13 @@ function(test_target_function)
     AMENT_DEPENDENCIES ${rmw_implementation}
   )
 
-  rcl_add_custom_gtest(test_get_node_names${target_suffix}
-    SRCS rcl/test_get_node_names.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    LIBRARIES ${PROJECT_NAME}
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp"
-  )
+  # rcl_add_custom_gtest(test_get_node_names${target_suffix}
+  #   SRCS rcl/test_get_node_names.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   LIBRARIES ${PROJECT_NAME}
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp"
+  # )
 
   rcl_add_custom_gtest(test_lexer${target_suffix}
     SRCS rcl/test_lexer.cpp
@@ -113,22 +113,22 @@ function(test_target_function)
     AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp"
   )
 
-  rcl_add_custom_gtest(test_graph${target_suffix}
-    SRCS rcl/test_graph.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    LIBRARIES ${PROJECT_NAME}
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
-    TIMEOUT 120
-  )
+  # rcl_add_custom_gtest(test_graph${target_suffix}
+  #   SRCS rcl/test_graph.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   LIBRARIES ${PROJECT_NAME}
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
+  #   TIMEOUT 120
+  # )
 
-  rcl_add_custom_gtest(test_info_by_topic${target_suffix}
-    SRCS rcl/test_info_by_topic.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    LIBRARIES ${PROJECT_NAME} wait_for_entity_helpers
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
-  )
+  # rcl_add_custom_gtest(test_info_by_topic${target_suffix}
+  #   SRCS rcl/test_info_by_topic.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   LIBRARIES ${PROJECT_NAME} wait_for_entity_helpers
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
+  # )
 
   rcl_add_custom_gtest(test_count_matched${target_suffix}
     SRCS rcl/test_count_matched.cpp
@@ -217,27 +217,27 @@ function(test_target_function)
     AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
   )
 
-  rcl_add_custom_gtest(test_publisher_wait_all_ack${target_suffix}
-    SRCS rcl/test_publisher_wait_all_ack.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../src/rcl/
-    LIBRARIES ${PROJECT_NAME} mimick wait_for_entity_helpers
-    AMENT_DEPENDENCIES ${rmw_implementation}
-      "osrf_testing_tools_cpp"
-      "rcpputils"
-      "rcutils"
-      "rosidl_runtime_c"
-      "test_msgs"
-  )
-
-  rcl_add_custom_gtest(test_service${target_suffix}
-    SRCS rcl/test_service.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    LIBRARIES ${PROJECT_NAME} mimick wait_for_entity_helpers
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
-  )
+  # rcl_add_custom_gtest(test_publisher_wait_all_ack${target_suffix}
+  #   SRCS rcl/test_publisher_wait_all_ack.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../src/rcl/
+  #   LIBRARIES ${PROJECT_NAME} mimick wait_for_entity_helpers
+  #   AMENT_DEPENDENCIES ${rmw_implementation}
+  #     "osrf_testing_tools_cpp"
+  #     "rcpputils"
+  #     "rcutils"
+  #     "rosidl_runtime_c"
+  #     "test_msgs"
+  # )
+
+  # rcl_add_custom_gtest(test_service${target_suffix}
+  #   SRCS rcl/test_service.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   LIBRARIES ${PROJECT_NAME} mimick wait_for_entity_helpers
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
+  # )
 
   rcl_add_custom_gtest(test_subscription${target_suffix}
     SRCS rcl/test_subscription.cpp
@@ -255,8 +255,8 @@ function(test_target_function)
     message(STATUS "Enabling message timestamp test for ${rmw_implementation}")
     target_compile_definitions(test_subscription${target_suffix}
       PUBLIC "RMW_TIMESTAMPS_SUPPORTED=1" "RMW_RECEIVED_TIMESTAMP_SUPPORTED=1")
-    target_compile_definitions(test_service${target_suffix}
-      PUBLIC "RMW_TIMESTAMPS_SUPPORTED=1" "RMW_RECEIVED_TIMESTAMP_SUPPORTED=1")
+    # target_compile_definitions(test_service${target_suffix}
+    #   PUBLIC "RMW_TIMESTAMPS_SUPPORTED=1" "RMW_RECEIVED_TIMESTAMP_SUPPORTED=1")
   else()
     if(rmw_implementation STREQUAL "rmw_cyclonedds_cpp")
       message(STATUS "Enabling only source timestamp test for ${rmw_implementation}")
@@ -269,14 +269,14 @@ function(test_target_function)
     endif()
   endif()
 
-  rcl_add_custom_gtest(test_events${target_suffix}
-    SRCS rcl/test_events.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../src/rcl/
-    LIBRARIES ${PROJECT_NAME}
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
-  )
+  # rcl_add_custom_gtest(test_events${target_suffix}
+  #   SRCS rcl/test_events.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../src/rcl/
+  #   LIBRARIES ${PROJECT_NAME}
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "test_msgs"
+  # )
 
   rcl_add_custom_gtest(test_wait${target_suffix}
     SRCS rcl/test_wait.cpp
@@ -286,13 +286,13 @@ function(test_target_function)
     AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp"
   )
 
-  rcl_add_custom_gtest(test_logging_rosout${target_suffix}
-    SRCS rcl/test_logging_rosout.cpp
-    ENV ${rmw_implementation_env_var}
-    APPEND_LIBRARY_DIRS ${extra_lib_dirs}
-    LIBRARIES ${PROJECT_NAME}
-    AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "rcl_interfaces"
-  )
+  # rcl_add_custom_gtest(test_logging_rosout${target_suffix}
+  #   SRCS rcl/test_logging_rosout.cpp
+  #   ENV ${rmw_implementation_env_var}
+  #   APPEND_LIBRARY_DIRS ${extra_lib_dirs}
+  #   LIBRARIES ${PROJECT_NAME}
+  #   AMENT_DEPENDENCIES ${rmw_implementation} "osrf_testing_tools_cpp" "rcl_interfaces"
+  # )
 
   rcl_add_custom_gtest(test_namespace${target_suffix}
     SRCS test_namespace.cpp
