diff --git a/test/cmake/rcl_add_custom_gtest.cmake b/test/cmake/rcl_add_custom_gtest.cmake
index 0ebe4bd..9773c12 100644
--- a/test/cmake/rcl_add_custom_gtest.cmake
+++ b/test/cmake/rcl_add_custom_gtest.cmake
@@ -74,7 +74,7 @@ macro(rcl_add_custom_gtest target)
   endif()
 
   # Pass args along to ament_add_gtest().
-  ament_add_gtest(${target} ${_ARG_SRCS} ${_ARG_ENV} ${_ARG_APPEND_ENV} ${_ARG_APPEND_LIBRARY_DIRS}
+  ament_add_ros_isolated_gtest(${target} ${_ARG_SRCS} ${_ARG_ENV} ${_ARG_APPEND_ENV} ${_ARG_APPEND_LIBRARY_DIRS}
                   ${_ARG_SKIP_TEST} ${_ARG_TIMEOUT})
   # Check if the target was actually created.
   if(TARGET ${target})
