diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d413505..80517300 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,14 +113,15 @@ if(BUILD_TESTING)
     set(append_library_dirs "${append_library_dirs}/$<CONFIG>")
   endif()
 
-  ament_add_gtest(test_component_manager test/test_component_manager.cpp
+  find_package(ament_cmake_ros)
+  ament_add_ros_isolated_gtest(test_component_manager test/test_component_manager.cpp
     APPEND_ENV AMENT_PREFIX_PATH=${CMAKE_CURRENT_BINARY_DIR}/test_ament_index/$<CONFIG>
     APPEND_LIBRARY_DIRS "${append_library_dirs}")
   if(TARGET test_component_manager)
     target_link_libraries(test_component_manager component_manager)
   endif()
 
-  ament_add_gtest(test_component_manager_api test/test_component_manager_api.cpp
+  ament_add_ros_isolated_gtest(test_component_manager_api test/test_component_manager_api.cpp
     APPEND_ENV AMENT_PREFIX_PATH=${CMAKE_CURRENT_BINARY_DIR}/test_ament_index/$<CONFIG>
     APPEND_LIBRARY_DIRS "${append_library_dirs}")
   if(TARGET test_component_manager_api)
