diff --git a/include/dynamic_reconfigure/server.h b/include/dynamic_reconfigure/server.h
index dac2b91..32ad127 100644
--- a/include/dynamic_reconfigure/server.h
+++ b/include/dynamic_reconfigure/server.h
@@ -218,7 +218,7 @@ private:
     ConfigType new_config = config_;
     new_config.__fromMessage__(req.config);
     new_config.__clamp__();
-    uint32_t level = config_.__level__(new_config);
+    const uint32_t level = config_.__level__(new_config);
 
     callCallback(new_config, level);
 
