diff --git a/plugins/input_at_waypoint.cpp b/plugins/input_at_waypoint.cpp
index 5c613219..00a4ed8c 100644
--- a/plugins/input_at_waypoint.cpp
+++ b/plugins/input_at_waypoint.cpp
@@ -49,7 +49,7 @@ void InputAtWaypoint::initialize(
   logger_ = node->get_logger();
   clock_ = node->get_clock();
 
-  double timeout;
+  double timeout = 0.0;
   std::string input_topic;
   nav2_util::declare_parameter_if_not_declared(
     node, plugin_name + ".timeout",
