diff --git a/include/pcl_ros/point_cloud.h b/include/pcl_ros/point_cloud.h
index 9044d77..9553fd0 100644
--- a/include/pcl_ros/point_cloud.h
+++ b/include/pcl_ros/point_cloud.h
@@ -3,7 +3,7 @@
 
 #include <ros/ros.h>
 #include <pcl/point_cloud.h>
-#include <pcl/point_traits.h>
+#include <pcl/type_traits.h>
 #include <pcl/for_each_type.h>
 #include <pcl/conversions.h>
 #include <pcl_conversions/pcl_conversions.h>
