diff --git a/setup.cfg b/setup.cfg
index e03c5c2..2313625 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -30,7 +30,7 @@ install_requires =
 packages = find:
 tests_require =
   flake8>=3.6.0
-  flake8-blind-except
+  flake8-blind-except<0.2.0
   flake8-builtins
   flake8-class-newline
   flake8-comprehensions
@@ -39,6 +39,7 @@ tests_require =
   flake8-import-order
   flake8-quotes
   pep8-naming
+  pydocstyle<6.0.0
   pylint
   pytest
   pytest-cov
@@ -48,6 +49,7 @@ zip_safe = true
 [tool:pytest]
 filterwarnings =
     error
+    ignore:lib2to3 package is deprecated::scspell
 junit_suite_name = colcon-package-information
 
 [options.entry_points]
@@ -60,3 +62,4 @@ colcon_core.verb =
 
 [flake8]
 import-order-style = google
+ignore=E741,W504
