diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0ec5a2..d8f64c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,6 +166,10 @@ endif()
 if(BUILD_TESTING)
   set(ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS ${LTTNG_INCLUDE_DIRS})
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   ament_lint_auto_find_test_dependencies()
 
   if(TRACETOOLS_STATUS_CHECKING_TOOL)
