diff --git a/setup.cfg b/setup.cfg
index 78fb170..54ca773 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -42,7 +42,7 @@ zip_safe = false
 
 [options.extras_require]
 test =
-  flake8>=3.6.0
+  flake8>=3.6.0,<6.0.0
   flake8-blind-except
   flake8-builtins
   flake8-class-newline
@@ -52,6 +52,7 @@ test =
   flake8-import-order
   flake8-quotes
   pep8-naming
+  pydocstyle<=6.1.1
   pylint
   pytest
   pytest-cov
