diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 6ff732b3..2cfc2fb7 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -129,9 +129,9 @@ target_link_libraries(test_goal_intent_search
 )
 
 # Test corner smoothing
-ament_add_gtest(test_corner_smoothing
-  test_corner_smoothing.cpp
-)
-target_link_libraries(test_corner_smoothing
-  ${library_name}
-)
+#ament_add_gtest(test_corner_smoothing
+#  test_corner_smoothing.cpp
+#)
+#target_link_libraries(test_corner_smoothing
+#  ${library_name}
+#)
