diff --git a/Tests/CMakeLists.txt b/Tests/CMakeLists.txt
index bc594e7..c8ffb90 100644
--- a/Tests/CMakeLists.txt
+++ b/Tests/CMakeLists.txt
@@ -64,7 +64,7 @@ if (OGRE_BUILD_TESTS)
       add_definitions(-Wno-undef)
     endif()
     add_executable(Test_Ogre ${HEADER_FILES} ${SOURCE_FILES} ${RESOURCE_FILES} )
-    ogre_install_target(Test_Ogre "" FALSE)
+    #ogre_install_target(Test_Ogre "" FALSE)
     target_link_libraries(Test_Ogre OgreBites Codec_STBI ${OGRE_LIBRARIES} GTest::gtest)
     
     if(ANDROID)
