diff --git a/include/ros/node_handle.h b/include/ros/node_handle.h
index 845292c84..61e81be54 100644
--- a/include/ros/node_handle.h
+++ b/include/ros/node_handle.h
@@ -28,6 +28,8 @@
 #ifndef ROSCPP_NODE_HANDLE_H
 #define ROSCPP_NODE_HANDLE_H
 
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
+
 #include "ros/forwards.h"
 #include "ros/publisher.h"
 #include "ros/subscriber.h"
diff --git a/include/ros/publisher.h b/include/ros/publisher.h
index c5d1b71bf..12bc7adb4 100644
--- a/include/ros/publisher.h
+++ b/include/ros/publisher.h
@@ -28,6 +28,8 @@
 #ifndef ROSCPP_PUBLISHER_HANDLE_H
 #define ROSCPP_PUBLISHER_HANDLE_H
 
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
+
 #include "ros/forwards.h"
 #include "ros/common.h"
 #include "ros/message.h"
