diff --git a/include/ros/publisher.h b/include/ros/publisher.h
index c5d1b71bf..ca84e7cc5 100644
--- a/include/ros/publisher.h
+++ b/include/ros/publisher.h
@@ -80,10 +80,10 @@ namespace ros
         if (impl_->md5sum_ == "*" ||
             std::string(mt::md5sum<M>(*message)) == "*" ||
             impl_->md5sum_ == mt::md5sum<M>(*message)) {
-          ROS_DEBUG_ONCE("Trying to publish message of type [%s/%s] on a "
-                    "publisher with type [%s/%s]",
-                    mt::datatype<M>(*message), mt::md5sum<M>(*message),
-                    impl_->datatype_.c_str(), impl_->md5sum_.c_str());
+          // ROS_DEBUG_ONCE("Trying to publish message of type [%s/%s] on a "
+          //           "publisher with type [%s/%s]",
+          //           mt::datatype<M>(*message), mt::md5sum<M>(*message),
+          //           impl_->datatype_.c_str(), impl_->md5sum_.c_str());
         }
 
       SerializedMessage m;
@@ -116,10 +116,10 @@ namespace ros
         if (impl_->md5sum_ == "*" ||
             std::string(mt::md5sum<M>(message)) == "*" ||
             impl_->md5sum_ == mt::md5sum<M>(message)) {
-          ROS_DEBUG_ONCE("Trying to publish message of type [%s/%s] on a "
-                    "publisher with type [%s/%s]",
-                    mt::datatype<M>(message), mt::md5sum<M>(message),
-                    impl_->datatype_.c_str(), impl_->md5sum_.c_str());
+          // ROS_DEBUG_ONCE("Trying to publish message of type [%s/%s] on a "
+          //           "publisher with type [%s/%s]",
+          //           mt::datatype<M>(message), mt::md5sum<M>(message),
+          //           impl_->datatype_.c_str(), impl_->md5sum_.c_str());
         }
 
       SerializedMessage m;
diff --git a/include/ros/subscription_callback_helper.h b/include/ros/subscription_callback_helper.h
index fca3b5f44..7941ca29c 100644
--- a/include/ros/subscription_callback_helper.h
+++ b/include/ros/subscription_callback_helper.h
@@ -123,7 +123,7 @@ public:
 
     if (!msg)
     {
-      ROS_DEBUG("Allocation failed for message of type [%s]", getTypeInfo().name());
+      // ROS_DEBUG("Allocation failed for message of type [%s]", getTypeInfo().name());
       return VoidConstPtr();
     }
 
