diff --git a/speech_tools/config/compilers/gcc_defaults.mak b/speech_tools/config/compilers/gcc_defaults.mak
index b66e5bc..673206a 100644
--- a/speech_tools/config/compilers/gcc_defaults.mak
+++ b/speech_tools/config/compilers/gcc_defaults.mak
@@ -62,8 +62,8 @@ VERBOSE_CCFLAGS   =
 VERBOSE_CXXFLAGS  = 
 VERBOSE_LINKFLAGS =
 
-OPTIMISE_CCFLAGS   = -O$(OPTIMISE)
-OPTIMISE_CXXFLAGS  = -O$(OPTIMISE)
-OPTIMISE_LINKFLAGS = -O$(OPTIMISE)
+OPTIMISE_CCFLAGS   = -O$(OPTIMISE) -g -std=gnu++98 -fno-strict-aliasing -fno-delete-null-pointer-checks
+OPTIMISE_CXXFLAGS  = -O$(OPTIMISE) -g -std=gnu++98 -fno-strict-aliasing -fno-delete-null-pointer-checks
+OPTIMISE_LINKFLAGS = -O$(OPTIMISE) -g
 
 PROFILE_DEFAULT = gprof
 
diff --git a/speech_tools/config/config.in b/speech_tools/config/config.in
index 1104720..92ba8ec 100644
--- a/speech_tools/config/config.in
+++ b/speech_tools/config/config.in
@@ -56,7 +56,7 @@ CONFIGURATION =
 ##
 ## OPTIMISE=4 will turn off DEBUG
 
-OPTIMISE=3
+OPTIMISE=s
 WARN=1
 # VERBOSE=1
 #DEBUG=1
