diff --git a/include/mcl_3dl/noise_generators/diagonal_noise_generator.h b/include/mcl_3dl/noise_generators/diagonal_noise_generator.h
index 9459284..2ccf9d3 100644
--- a/include/mcl_3dl/noise_generators/diagonal_noise_generator.h
+++ b/include/mcl_3dl/noise_generators/diagonal_noise_generator.h
@@ -66,6 +66,11 @@ public:
     std::vector<FLT_TYPE> noise(sigma_.size());
     for (size_t i = 0; i < sigma_.size(); i++)
     {
+      if (sigma_[i] == 0)
+      {
+        noise[i] = Parent::mean_[i];
+        continue;
+      }
       std::normal_distribution<FLT_TYPE> nd(Parent::mean_[i], sigma_[i]);
       noise[i] = nd(engine);
     }
