diff --git a/CMakeLists.txt b/CMakeLists.txt
index a517ed3..5938b38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,6 @@ if(BUILD_TESTING)
   find_package(ament_cmake_cpplint REQUIRED)
   find_package(ament_cmake_gtest REQUIRED)
   find_package(ament_cmake_lint_cmake REQUIRED)
-  find_package(ament_cmake_uncrustify REQUIRED)
   find_package(ament_cmake_xmllint REQUIRED)
 
   set(
@@ -70,10 +69,6 @@ if(BUILD_TESTING)
   )
   ament_cpplint(EXCLUDE ${_linter_excludes})
   ament_lint_cmake()
-  ament_uncrustify(
-    EXCLUDE ${_linter_excludes}
-    LANGUAGE c++
-  )
   ament_xmllint()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
