diff --git a/source/posix_wrapper/mutex.cpp b/source/posix_wrapper/mutex.cpp
index d739dc38d..66580471c 100644
--- a/source/posix_wrapper/mutex.cpp
+++ b/source/posix_wrapper/mutex.cpp
@@ -36,7 +36,11 @@ mutex::mutex(bool f_isRecursive) noexcept
                           .evaluate()
                           .has_error();
     isInitialized &=
+#if defined(PTHREAD_MUTEX_RECURSIVE_NP) && defined(PTHREAD_MUTEX_FAST_NP)
         !posixCall(pthread_mutexattr_settype)(&attr, f_isRecursive ? PTHREAD_MUTEX_RECURSIVE_NP : PTHREAD_MUTEX_FAST_NP)
+#else
+        !posixCall(pthread_mutexattr_settype)(&attr, f_isRecursive ? PTHREAD_MUTEX_RECURSIVE : PTHREAD_MUTEX_NORMAL)
+#endif
              .returnValueMatchesErrno()
              .evaluate()
              .has_error();
