diff --git a/setup.cfg b/setup.cfg
index 82577a7..a0b1760 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -48,6 +48,7 @@ zip_safe = true
 [tool:pytest]
 filterwarnings =
     error
+    ignore:lib2to3 package is deprecated::scspell
 junit_suite_name = colcon-recursive-crawl
 
 [options.entry_points]
@@ -56,3 +57,4 @@ colcon_core.package_discovery =
 
 [flake8]
 import-order-style = google
+ignore=E741,W504
