diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 77245e4..59830ae 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -57,15 +57,15 @@ ament_add_gtest_executable(test_context
 )
 target_link_libraries(test_context ${PROJECT_NAME} mimick osrf_testing_tools_cpp::memory_tools)
 
-ament_add_gtest_executable(test_get_node_names
-  rcl/test_get_node_names.cpp
-)
-target_link_libraries(test_get_node_names ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools)
+# ament_add_gtest_executable(test_get_node_names
+#   rcl/test_get_node_names.cpp
+# )
+# target_link_libraries(test_get_node_names ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools)
 
-ament_add_gtest_executable(test_graph
-  rcl/test_graph.cpp
-)
-target_link_libraries(test_graph ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools ${test_msgs_TARGETS})
+# ament_add_gtest_executable(test_graph
+#   rcl/test_graph.cpp
+# )
+# target_link_libraries(test_graph ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools ${test_msgs_TARGETS})
 
 ament_add_gtest_executable(test_info_by_topic
   rcl/test_info_by_topic.cpp
@@ -197,16 +197,18 @@ function(test_target)
     ENV ${rmw_implementation_env_var} ${memory_tools_ld_preload_env_var}
   )
 
-  ament_add_gtest_test(test_get_node_names
-    TEST_NAME test_get_node_names${target_suffix}
+  ament_add_ros_isolated_gtest(test_get_node_names${target_suffix}
+    rcl/test_get_node_names.cpp
     ENV ${rmw_implementation_env_var}
   )
+  target_link_libraries(test_get_node_names${target_suffix} ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools)
 
-  ament_add_gtest_test(test_graph
-    TEST_NAME test_graph${target_suffix}
+  ament_add_ros_isolated_gtest(test_graph${target_suffix}
+    rcl/test_graph.cpp
     ENV ${rmw_implementation_env_var}
     TIMEOUT 120
   )
+  target_link_libraries(test_graph${target_suffix} ${PROJECT_NAME} osrf_testing_tools_cpp::memory_tools ${test_msgs_TARGETS})
 
   ament_add_gtest_test(test_info_by_topic
     TEST_NAME test_info_by_topic${target_suffix}
