diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bbf1f5..b8d87a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,8 +30,8 @@ include_directories(include ${catkin_INCLUDE_DIRS})
 #############
 
 if(CATKIN_ENABLE_TESTING)
-  catkin_add_gmock(controller_base_test test/controller_base_test.cpp)
-  target_link_libraries(controller_base_test ${catkin_LIBRARIES})
+#  catkin_add_gmock(controller_base_test test/controller_base_test.cpp)
+#  target_link_libraries(controller_base_test ${catkin_LIBRARIES})
 endif()
 
 
