diff --git a/CHANGELOG.rst b/CHANGELOG.rst
index bed3c7d..50fd526 100644
--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -2,6 +2,18 @@
 Changelog for package pcl_ros
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+Forthcoming
+-----------
+* Adding separate leaf sizes for each axis in the VoxelGrid filter (`#393 <https://github.com/ros-perception/perception_pcl/issues/393>`_)
+* Add logic for missing pcl/point_traits.h in newer PCL versions (`#381 <https://github.com/ros-perception/perception_pcl/issues/381>`_)
+  point_traits.h is no longer available, starting with PCL 1.13.0
+  Since PCL 1.11.0, type_traits.h should be used instead
+* Add boost include, missing in upcoming PCL versions (`#373 <https://github.com/ros-perception/perception_pcl/issues/373>`_)
+  Was removed in pcl/conversions.h here: https://github.com/PointCloudLibrary/pcl/commit/292593abd3b69af315c7fe3379363bdce7800d5a
+* Switch to hpp headers of pluginlib (`#370 <https://github.com/ros-perception/perception_pcl/issues/370>`_)
+  Co-authored-by: Jochen Sprickerhof <jspricke@debian.org>
+* Contributors: Jochen Sprickerhof, Markus Vieth, Tom Moore
+
 1.7.0 (2019-07-03)
 ------------------
 * Added option for fixed filename (via local parameter) (`#102 <https://github.com/ros-perception/perception_pcl/issues/102>`_)
diff --git a/cfg/VoxelGrid.cfg b/cfg/VoxelGrid.cfg
index 07dd2c3..5d474ce 100755
--- a/cfg/VoxelGrid.cfg
+++ b/cfg/VoxelGrid.cfg
@@ -13,7 +13,11 @@ from dynamic_reconfigure.parameter_generator_catkin import *
 gen = ParameterGenerator ()
 # def add (self, name, paramtype, level, description, default = None, min = None, max = None, edit_method = ""):
 gen.add ("leaf_size", double_t, 0, "The size of a leaf (on x,y,z) used for downsampling.", 0.01, 0, 1.0)
+gen.add ("leaf_size_x", double_t, 0, "The size of a leaf (on x) used for downsampling. If negative, leaf_size is used instead.", -1.0, -1.0, 1.0)
+gen.add ("leaf_size_y", double_t, 0, "The size of a leaf (on y) used for downsampling. If negative, leaf_size is used instead.", -1.0, -1.0, 1.0)
+gen.add ("leaf_size_z", double_t, 0, "The size of a leaf (on z) used for downsampling. If negative, leaf_size is used instead.", -1.0, -1.0, 1.0)
 gen.add ("min_points_per_voxel", int_t, 0, "The minimum number of points required for a voxel to be used.", 1, 1, 100000)
+
 add_common_parameters (gen)
 
 exit (gen.generate (PACKAGE, "pcl_ros", "VoxelGrid"))
diff --git a/include/pcl_ros/point_cloud.h b/include/pcl_ros/point_cloud.h
index 1e03827..ea91f75 100644
--- a/include/pcl_ros/point_cloud.h
+++ b/include/pcl_ros/point_cloud.h
@@ -3,11 +3,17 @@
 
 #include <ros/ros.h>
 #include <pcl/point_cloud.h>
+#include <pcl/pcl_config.h> // for PCL_VERSION_COMPARE
+#if PCL_VERSION_COMPARE(>=, 1, 11, 0)
+#include <pcl/type_traits.h>
+#else
 #include <pcl/point_traits.h>
+#endif // PCL_VERSION_COMPARE(>=, 1, 11, 0)
 #include <pcl/for_each_type.h>
 #include <pcl/conversions.h>
 #include <pcl_conversions/pcl_conversions.h>
 #include <sensor_msgs/PointCloud2.h>
+#include <boost/foreach.hpp> // for BOOST_FOREACH
 #include <boost/mpl/size.hpp>
 #include <boost/ref.hpp>
 #include <boost/thread/mutex.hpp>
@@ -281,7 +287,6 @@ namespace ros
 #include <type_traits>  // for std::is_same
 #include <memory>       // for std::shared_ptr
 
-#include <pcl/pcl_config.h>
 #if PCL_VERSION_COMPARE(>=, 1, 11, 0)
 #include <pcl/memory.h>
 #elif PCL_VERSION_COMPARE(>=, 1, 10, 0)
diff --git a/src/pcl_ros/features/boundary.cpp b/src/pcl_ros/features/boundary.cpp
index 26ee07c..61ae786 100644
--- a/src/pcl_ros/features/boundary.cpp
+++ b/src/pcl_ros/features/boundary.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/boundary.h"
 
 void
diff --git a/src/pcl_ros/features/feature.cpp b/src/pcl_ros/features/feature.cpp
index 8647e4e..cee3dbc 100644
--- a/src/pcl_ros/features/feature.cpp
+++ b/src/pcl_ros/features/feature.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-//#include <pluginlib/class_list_macros.h>
+//#include <pluginlib/class_list_macros.hpp>
 // Include the implementations here instead of compiling them separately to speed up compile time
 //#include "normal_3d.cpp"
 //#include "boundary.cpp"
diff --git a/src/pcl_ros/features/fpfh.cpp b/src/pcl_ros/features/fpfh.cpp
index 53be549..f49b05e 100644
--- a/src/pcl_ros/features/fpfh.cpp
+++ b/src/pcl_ros/features/fpfh.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/fpfh.h"
 
 void 
diff --git a/src/pcl_ros/features/fpfh_omp.cpp b/src/pcl_ros/features/fpfh_omp.cpp
index e4adcab..c811a40 100644
--- a/src/pcl_ros/features/fpfh_omp.cpp
+++ b/src/pcl_ros/features/fpfh_omp.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/fpfh_omp.h"
 
 void 
diff --git a/src/pcl_ros/features/moment_invariants.cpp b/src/pcl_ros/features/moment_invariants.cpp
index a6e2249..52801ff 100644
--- a/src/pcl_ros/features/moment_invariants.cpp
+++ b/src/pcl_ros/features/moment_invariants.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/moment_invariants.h"
 
 void 
diff --git a/src/pcl_ros/features/normal_3d.cpp b/src/pcl_ros/features/normal_3d.cpp
index 042186a..3ca08fb 100644
--- a/src/pcl_ros/features/normal_3d.cpp
+++ b/src/pcl_ros/features/normal_3d.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/normal_3d.h"
 
 void 
diff --git a/src/pcl_ros/features/normal_3d_omp.cpp b/src/pcl_ros/features/normal_3d_omp.cpp
index 3e92d2f..9653507 100644
--- a/src/pcl_ros/features/normal_3d_omp.cpp
+++ b/src/pcl_ros/features/normal_3d_omp.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/normal_3d_omp.h"
 
 void 
diff --git a/src/pcl_ros/features/normal_3d_tbb.cpp b/src/pcl_ros/features/normal_3d_tbb.cpp
index 680a4a0..8526176 100644
--- a/src/pcl_ros/features/normal_3d_tbb.cpp
+++ b/src/pcl_ros/features/normal_3d_tbb.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/normal_3d_tbb.h"
 
 #if defined HAVE_TBB
diff --git a/src/pcl_ros/features/pfh.cpp b/src/pcl_ros/features/pfh.cpp
index dd8409e..b00cadc 100644
--- a/src/pcl_ros/features/pfh.cpp
+++ b/src/pcl_ros/features/pfh.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/pfh.h"
 
 void 
diff --git a/src/pcl_ros/features/principal_curvatures.cpp b/src/pcl_ros/features/principal_curvatures.cpp
index 501d686..f8fe44c 100644
--- a/src/pcl_ros/features/principal_curvatures.cpp
+++ b/src/pcl_ros/features/principal_curvatures.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/principal_curvatures.h"
 
 void 
diff --git a/src/pcl_ros/features/shot.cpp b/src/pcl_ros/features/shot.cpp
index ed6ba44..085855f 100644
--- a/src/pcl_ros/features/shot.cpp
+++ b/src/pcl_ros/features/shot.cpp
@@ -34,7 +34,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/shot.h"
 
 void 
diff --git a/src/pcl_ros/features/shot_omp.cpp b/src/pcl_ros/features/shot_omp.cpp
index 4563f12..eea7005 100644
--- a/src/pcl_ros/features/shot_omp.cpp
+++ b/src/pcl_ros/features/shot_omp.cpp
@@ -34,7 +34,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/shot_omp.h"
 
 void 
diff --git a/src/pcl_ros/features/vfh.cpp b/src/pcl_ros/features/vfh.cpp
index ece448f..e1f6a40 100644
--- a/src/pcl_ros/features/vfh.cpp
+++ b/src/pcl_ros/features/vfh.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/features/vfh.h"
 
 void 
diff --git a/src/pcl_ros/filters/crop_box.cpp b/src/pcl_ros/filters/crop_box.cpp
index 2392e2b..af173dc 100644
--- a/src/pcl_ros/filters/crop_box.cpp
+++ b/src/pcl_ros/filters/crop_box.cpp
@@ -36,7 +36,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/crop_box.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/extract_indices.cpp b/src/pcl_ros/filters/extract_indices.cpp
index e92407a..89b1608 100644
--- a/src/pcl_ros/filters/extract_indices.cpp
+++ b/src/pcl_ros/filters/extract_indices.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/extract_indices.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/passthrough.cpp b/src/pcl_ros/filters/passthrough.cpp
index 110fbac..f85f536 100644
--- a/src/pcl_ros/filters/passthrough.cpp
+++ b/src/pcl_ros/filters/passthrough.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/passthrough.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/project_inliers.cpp b/src/pcl_ros/filters/project_inliers.cpp
index 704dc48..6781d26 100644
--- a/src/pcl_ros/filters/project_inliers.cpp
+++ b/src/pcl_ros/filters/project_inliers.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/project_inliers.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/radius_outlier_removal.cpp b/src/pcl_ros/filters/radius_outlier_removal.cpp
index 331c648..4b242db 100644
--- a/src/pcl_ros/filters/radius_outlier_removal.cpp
+++ b/src/pcl_ros/filters/radius_outlier_removal.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/radius_outlier_removal.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/statistical_outlier_removal.cpp b/src/pcl_ros/filters/statistical_outlier_removal.cpp
index 2085dee..fa69a75 100644
--- a/src/pcl_ros/filters/statistical_outlier_removal.cpp
+++ b/src/pcl_ros/filters/statistical_outlier_removal.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/statistical_outlier_removal.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/filters/voxel_grid.cpp b/src/pcl_ros/filters/voxel_grid.cpp
index a3bf77f..389df97 100644
--- a/src/pcl_ros/filters/voxel_grid.cpp
+++ b/src/pcl_ros/filters/voxel_grid.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/filters/voxel_grid.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////
@@ -74,14 +74,17 @@ pcl_ros::VoxelGrid::config_callback (pcl_ros::VoxelGridConfig &config, uint32_t
   boost::mutex::scoped_lock lock (mutex_);
 
   Eigen::Vector3f leaf_size = impl_.getLeafSize ();
+  Eigen::Vector3f config_leaf_size;
 
-  if (leaf_size[0] != config.leaf_size)
-  {
-    leaf_size.setConstant (config.leaf_size);
-    NODELET_DEBUG ("[config_callback] Setting the downsampling leaf size to: %f.", leaf_size[0]);
-    impl_.setLeafSize (leaf_size[0], leaf_size[1], leaf_size[2]);
+  config_leaf_size[0] = (config.leaf_size_x >= 0.0 ? config.leaf_size_x : config.leaf_size);
+  config_leaf_size[1] = (config.leaf_size_y >= 0.0 ? config.leaf_size_y : config.leaf_size);
+  config_leaf_size[2] = (config.leaf_size_z >= 0.0 ? config.leaf_size_z : config.leaf_size);
+
+  if (leaf_size != config_leaf_size) {
+    NODELET_DEBUG("[config_callback] Setting the downsampling leaf size to: (%f, %f, %f).",
+      config_leaf_size[0], config_leaf_size[1], config_leaf_size[2]);
+    impl_.setLeafSize(config_leaf_size[0], config_leaf_size[1], config_leaf_size[2]);
   }
-  
   unsigned int minPointsPerVoxel = impl_.getMinimumPointsNumberPerVoxel ();
   if (minPointsPerVoxel != ((unsigned int) config.min_points_per_voxel))
   {
diff --git a/src/pcl_ros/io/bag_io.cpp b/src/pcl_ros/io/bag_io.cpp
index 4209d9a..902ce2f 100644
--- a/src/pcl_ros/io/bag_io.cpp
+++ b/src/pcl_ros/io/bag_io.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include <ros/node_handle.h>
 #include "pcl_ros/io/bag_io.h"
 
diff --git a/src/pcl_ros/io/concatenate_data.cpp b/src/pcl_ros/io/concatenate_data.cpp
index 458c6bf..b0425ff 100644
--- a/src/pcl_ros/io/concatenate_data.cpp
+++ b/src/pcl_ros/io/concatenate_data.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/transforms.h"
 #include "pcl_ros/io/concatenate_data.h"
 
diff --git a/src/pcl_ros/io/concatenate_fields.cpp b/src/pcl_ros/io/concatenate_fields.cpp
index b9567c3..8b6b3c8 100644
--- a/src/pcl_ros/io/concatenate_fields.cpp
+++ b/src/pcl_ros/io/concatenate_fields.cpp
@@ -37,7 +37,7 @@
 
 /** \author Radu Bogdan Rusu */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/io/concatenate_fields.h"
 
 #include <pcl_conversions/pcl_conversions.h>
diff --git a/src/pcl_ros/io/io.cpp b/src/pcl_ros/io/io.cpp
index a2dcd45..58ff115 100644
--- a/src/pcl_ros/io/io.cpp
+++ b/src/pcl_ros/io/io.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include <sensor_msgs/PointCloud2.h>
 #include <message_filters/subscriber.h>
 //#include <pcl_ros/subscriber.h>
diff --git a/src/pcl_ros/io/pcd_io.cpp b/src/pcl_ros/io/pcd_io.cpp
index 9adf688..31fca76 100644
--- a/src/pcl_ros/io/pcd_io.cpp
+++ b/src/pcl_ros/io/pcd_io.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include <pcl_ros/io/pcd_io.h>
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/segmentation/extract_clusters.cpp b/src/pcl_ros/segmentation/extract_clusters.cpp
index 5599b40..2d41b68 100644
--- a/src/pcl_ros/segmentation/extract_clusters.cpp
+++ b/src/pcl_ros/segmentation/extract_clusters.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include <pcl/PointIndices.h>
 #include "pcl_ros/segmentation/extract_clusters.h"
 
diff --git a/src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp b/src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp
index ff823b1..af445cf 100644
--- a/src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp
+++ b/src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/transforms.h"
 #include "pcl_ros/segmentation/extract_polygonal_prism_data.h"
 
diff --git a/src/pcl_ros/segmentation/sac_segmentation.cpp b/src/pcl_ros/segmentation/sac_segmentation.cpp
index bc7b97e..06b1361 100644
--- a/src/pcl_ros/segmentation/sac_segmentation.cpp
+++ b/src/pcl_ros/segmentation/sac_segmentation.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/segmentation/sac_segmentation.h"
 
 #include <pcl_conversions/pcl_conversions.h>
diff --git a/src/pcl_ros/segmentation/segment_differences.cpp b/src/pcl_ros/segmentation/segment_differences.cpp
index e397954..bde054b 100644
--- a/src/pcl_ros/segmentation/segment_differences.cpp
+++ b/src/pcl_ros/segmentation/segment_differences.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/segmentation/segment_differences.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/pcl_ros/surface/convex_hull.cpp b/src/pcl_ros/surface/convex_hull.cpp
index 7590388..6719ace 100644
--- a/src/pcl_ros/surface/convex_hull.cpp
+++ b/src/pcl_ros/surface/convex_hull.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/surface/convex_hull.h"
 #include <geometry_msgs/PolygonStamped.h>
 
diff --git a/src/pcl_ros/surface/moving_least_squares.cpp b/src/pcl_ros/surface/moving_least_squares.cpp
index 9fd2e34..46c6a39 100644
--- a/src/pcl_ros/surface/moving_least_squares.cpp
+++ b/src/pcl_ros/surface/moving_least_squares.cpp
@@ -35,7 +35,7 @@
  *
  */
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "pcl_ros/surface/moving_least_squares.h"
 //////////////////////////////////////////////////////////////////////////////////////////////
 void
