Ported from https://github.com/ros-drivers/rosserial/pull/523

diff --git a/src/rosserial_python/SerialClient.py b/src/rosserial_python/SerialClient.py
index 27d4d54..998b820 100644
--- a/src/rosserial_python/SerialClient.py
+++ b/src/rosserial_python/SerialClient.py
@@ -269,7 +269,8 @@ class RosSerialServer:
             for sub in client.subscribers.values():
                 sub.unregister()
             for srv in client.services.values():
-                srv.unregister()
+                if isinstance(srv, ServiceServer):
+                    srv.unregister()
 
     def startSocketServer(self, port, address):
         rospy.loginfo("starting ROS Serial Python Node serial_node-%r" % address)
