diff --git a/include/pluginlib/class_loader_imp.hpp b/include/pluginlib/class_loader_imp.hpp
index d45b591..cc30033 100644
--- a/include/pluginlib/class_loader_imp.hpp
+++ b/include/pluginlib/class_loader_imp.hpp
@@ -47,6 +47,8 @@
 #include <utility>
 #include <vector>
 
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
+
 #include "boost/algorithm/string.hpp"
 #include "boost/bind.hpp"
 #include "boost/filesystem.hpp"
