diff --git a/include/performance_test_fixture/performance_test_fixture.hpp b/include/performance_test_fixture/performance_test_fixture.hpp
index 6d73771..7edf338 100644
--- a/include/performance_test_fixture/performance_test_fixture.hpp
+++ b/include/performance_test_fixture/performance_test_fixture.hpp
@@ -67,12 +67,12 @@ private:
  * measuring timing
  */
 #define PERFORMANCE_TEST_FIXTURE_PAUSE_MEASUREMENTS(state, code) \
-  do { \
-    state.PauseTiming(); \
-    set_are_allocation_measurements_active(false); \
-    code; \
-    set_are_allocation_measurements_active(true); \
-    state.ResumeTiming(); \
-  } while (0)
+        do { \
+          state.PauseTiming(); \
+          set_are_allocation_measurements_active(false); \
+          code; \
+          set_are_allocation_measurements_active(true); \
+          state.ResumeTiming(); \
+        } while (0)
 
 #endif  // PERFORMANCE_TEST_FIXTURE__PERFORMANCE_TEST_FIXTURE_HPP_
