diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b9cda78..14ca7a4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,6 +88,10 @@ endif()
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   find_package(ament_cmake_google_benchmark REQUIRED)
   find_package(benchmark REQUIRED)
   # Give cppcheck hints about macro definitions coming from outside this package
