diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ccf4bc..4cf7b59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ install(
 
 # define the exe to convert
 add_executable(convert src/convert.cpp)
-target_link_libraries(convert ${PROJECT_NAME} ${rosconsole_LIBRARIES})
+target_link_libraries(convert ${PROJECT_NAME} ${YAML_CPP_LIBRARIES} ${rosconsole_LIBRARIES})
 
 install(
   TARGETS convert
