diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ccf4bc..723c533 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ add_library(${PROJECT_NAME}
 add_library(${PROJECT_NAME}_wrapper
   src/parse_wrapper.cpp)
 
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${YAML_CPP_LIBRARIES} ${Boost_FILESYSTEM_LIBRARY})
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} yaml-cpp ${Boost_FILESYSTEM_LIBRARY})
 target_link_libraries(${PROJECT_NAME}_wrapper ${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
 
 # Don't prepend wrapper library name with lib and add to Python libs.
