diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0258b1..3a673c5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ add_executable(mcl_3dl
   src/point_types.cpp
 )
 target_link_libraries(mcl_3dl ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(mcl_3dl ${catkin_EXPORTED_TARGETS})
+add_dependencies(mcl_3dl ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
 
 
 if(CATKIN_ENABLE_TESTING)
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index feb154a..8f75edb 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -27,6 +27,7 @@ target_compile_options(test_point_types PRIVATE ${UNIT_TEST_COMPILE_OPTIONS})
 catkin_add_gtest(test_point_cloud_random_sampler src/test_point_cloud_random_sampler.cpp)
 target_link_libraries(test_point_cloud_random_sampler ${catkin_LIBRARIES} ${PCL_LIBRARIES})
 target_compile_options(test_point_cloud_random_sampler PRIVATE ${UNIT_TEST_COMPILE_OPTIONS})
+add_dependencies(test_point_cloud_random_sampler ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
 
 catkin_add_gtest(test_quat src/test_quat.cpp)
 target_link_libraries(test_quat ${catkin_LIBRARIES})
@@ -88,6 +89,7 @@ add_rostest_gtest(test_beam_likelihood tests/beam_likelihood_rostest.test
     ../src/point_types.cpp
 )
 target_link_libraries(test_beam_likelihood ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+add_dependencies(test_beam_likelihood ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
 
 add_rostest_gtest(test_global_localization tests/global_localization_rostest.test
     src/test_global_localization.cpp)
@@ -113,6 +115,7 @@ add_rostest_gtest(test_parameters tests/parameters_rostest.test
     ../src/parameters.cpp
 )
 target_link_libraries(test_parameters ${catkin_LIBRARIES})
+add_dependencies(test_parameters ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
 
 if(MCL_3DL_EXTRA_TESTS OR "$ENV{MCL_3DL_EXTRA_TESTS}" STREQUAL "ON")
 
