diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8e73ba..25f8576 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,9 +36,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
 ament_export_targets(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
 
 if(BUILD_TESTING)
-  find_package(ament_lint_auto REQUIRED)
-  ament_lint_auto_find_test_dependencies()
-
+  find_package(ament_cmake_gtest)
   ament_add_gtest(${PROJECT_NAME}_utest test/utest.cpp)
   if(TARGET ${PROJECT_NAME}_utest)
     target_include_directories(${PROJECT_NAME}_utest PUBLIC include)
