diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52c504b..a217d95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -198,6 +198,7 @@ if(BUILD_TESTING)
     set(tutorial_tests_to_test ${tutorial_tests})
     list(APPEND tutorial_tests_to_test ${service_tutorial_tests})
 
+    set(INDEX 0)
     foreach(tutorial_test ${tutorial_tests_to_test})
       set(DEMO_NODES_CPP_EXPECTED_OUTPUT "")
       set(DEMO_NODES_CPP_EXECUTABLE "")
@@ -255,7 +256,9 @@ if(BUILD_TESTING)
         ENV
         RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation}
         RMW_IMPLEMENTATION=${rmw_implementation}
+        ROS_DOMAIN_ID=${INDEX}
       )
+      math(EXPR INDEX "${INDEX} + 1")
       foreach(executable ${exe_list})
         set_property(
           TEST test_tutorial_${exe_list_underscore}${target_suffix}
