diff --git a/CMake/vtkOpenGL.cmake b/CMake/vtkOpenGL.cmake
index f4177f4e..71fdfe04 100644
--- a/CMake/vtkOpenGL.cmake
+++ b/CMake/vtkOpenGL.cmake
@@ -75,7 +75,7 @@ if(VTK_CAN_DO_ONSCREEN)
   # used. Otherwise, if only doing OFFSCREEN, the GL API is provided by the
   # offscreen library be it EGL or OSMESA.
   find_package(OpenGL REQUIRED)
-  include_directories(SYSTEM ${OPENGL_INCLUDE_DIR})
+  include_directories(SYSTEM ${OPENGL_INCLUDE_DIR} ${GLEW_INCLUDE_DIRS})
   if(APPLE)
     # Remove the deprecated AGL framework found by FindOpenGL.cmake
     # (this is only required for CMake 3.4.1 and earlier)
@@ -102,6 +102,6 @@ function(vtk_opengl_link target)
     vtk_module_link_libraries(${target} LINK_PRIVATE ${EGL_LIBRARIES})
   endif()
   if(VTK_CAN_DO_ONSCREEN)
-    vtk_module_link_libraries(${target} LINK_PRIVATE ${OPENGL_LIBRARIES})
+    vtk_module_link_libraries(${target} LINK_PRIVATE ${OPENGL_LIBRARIES} ${GLEW_LIBRARY})
   endif()
 endfunction()
