diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8befc05b..d12e6008 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,7 +200,7 @@ else()
 endif()
 
 # default to not using the system GLEW as ours has fixes in it right now
-set(VTK_USE_SYSTEM_GLEW OFF CACHE BOOL "do not use a system glew" FORCE)
+# set(VTK_USE_SYSTEM_GLEW OFF CACHE BOOL "do not use a system glew" FORCE)
 
 set(VTK_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 
