diff --git a/test/test_roslogging.py b/test/test_roslogging.py
index 67fbd7554..65a6874f4 100644
--- a/test/test_roslogging.py
+++ b/test/test_roslogging.py
@@ -39,7 +39,6 @@ except ImportError:
 import sys
 
 import re
-from nose.tools import assert_regexp_matches
 import rosgraph.roslogging
 
 
@@ -124,7 +123,7 @@ try:
                 r'[0-9]*\.[0-9]*',
                 r'[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}',
             ])
-            assert_regexp_matches(lout.getvalue().splitlines()[i], log_out)
+            assert re.match(log_out, lout.getvalue().splitlines()[i]) is not None
 
 finally:
 
diff --git a/test/test_roslogging_user_logger.py b/test/test_roslogging_user_logger.py
index ab2981b78..7a860d939 100644
--- a/test/test_roslogging_user_logger.py
+++ b/test/test_roslogging_user_logger.py
@@ -43,8 +43,7 @@ try:
 except ImportError:
     from io import StringIO
 import sys
-
-from nose.tools import assert_regexp_matches
+import re
 
 import rosgraph.roslogging
 
@@ -141,7 +140,7 @@ def test_roslogging_user_logger():
             r'[0-9]*\.[0-9]*',
             r'[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}',
         ])
-        assert_regexp_matches(lout.getvalue().strip(), log_expected)
+        assert re.match(log_expected, lout.getvalue().strip()) is not None
 
     finally:
         # restoring default ros handler
