diff --git a/src/rosgraph/roslogging.py b/src/rosgraph/roslogging.py
index 9ecc12189..586069ff5 100644
--- a/src/rosgraph/roslogging.py
+++ b/src/rosgraph/roslogging.py
@@ -55,23 +55,18 @@ class RospyLogger(logging.getLoggerClass()):
         Find the stack frame of the caller so that we can note the source
         file name, line number, and function name with class name if possible.
         """
-        file_name, lineno, func_name = super(RospyLogger, self).findCaller(*args, **kwargs)[:3]
-        file_name = os.path.normcase(file_name)
-
         f = inspect.currentframe()
         if f is not None:
             f = f.f_back
+
         while hasattr(f, "f_code"):
-            # Search for the right frame using the data already found by parent class.
-            co = f.f_code
-            filename = os.path.normcase(co.co_filename)
-            if filename == file_name and f.f_lineno == lineno and co.co_name == func_name:
+            if f.f_code.co_name == '_log':
                 break
             if f.f_back:
                 f = f.f_back
 
         # Jump up two more frames, as the logger methods have been double wrapped.
-        if f is not None and f.f_back and f.f_code and f.f_code.co_name == '_base_logger':
+        if f is not None and f.f_back and f.f_code and f.f_code.co_name == '_log':
             f = f.f_back
             if f.f_back:
                 f = f.f_back
