diff --git a/test/launch_yaml/test_executable.py b/test/launch_yaml/test_executable.py
index 5c6ec2b..5bc8798 100644
--- a/test/launch_yaml/test_executable.py
+++ b/test/launch_yaml/test_executable.py
@@ -42,20 +42,20 @@ def test_executable():
     ld = parser.parse_description(root_entity)
     executable = ld.entities[0]
     cmd = [i[0].perform(None) for i in executable.cmd]
-    assert(
+    assert (
         cmd == ['ls', '-l', '-a', '-s'])
-    assert(executable.cwd[0].perform(None) == '/')
-    assert(executable.name[0].perform(None) == 'my_ls')
-    assert(executable.shell is True)
-    assert(executable.output[0].perform(None) == 'log')
+    assert (executable.cwd[0].perform(None) == '/')
+    assert (executable.name[0].perform(None) == 'my_ls')
+    assert (executable.shell is True)
+    assert (executable.output[0].perform(None) == 'log')
     key, value = executable.additional_env[0]
     key = key[0].perform(None)
     value = value[0].perform(None)
-    assert(key == 'var')
-    assert(value == '1')
+    assert (key == 'var')
+    assert (value == '1')
     ls = LaunchService()
     ls.include_launch_description(ld)
-    assert(0 == ls.run())
+    assert (0 == ls.run())
 
 
 if __name__ == '__main__':
