diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3451e72..d491417 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,9 @@ install(DIRECTORY resource
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_flake8
+  )
 
   # Disabling copyright test. The copyright used in this package does not conform to
   # ament's copyright tests
