diff --git a/CHANGELOG.rst b/theora_image_transport/CHANGELOG.rst
index f44dcdc..e68809b 100644
--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -2,6 +2,12 @@
 Changelog for package theora_image_transport
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+1.15.0 (2023-01-30)
+-------------------
+* Update maintainer in ros1 branches (`#132 <https://github.com/ros-perception/image_transport_plugins/issues/132>`_)
+* Fix copyright year 20012 -> 2012 (`#80 <https://github.com/ros-perception/image_transport_plugins/issues/80>`_)
+* Contributors: Kenji Brameld, Lucas Walter, Michael Carroll, ijnek, v4hn
+
 1.14.0 (2020-05-14)
 -------------------
 * Bump CMake version to avoid CMP0048 warning (`#53 <https://github.com/ros-perception/image_transport_plugins/issues/53>`_)
diff --git a/include/theora_image_transport/theora_publisher.h b/theora_image_transport/include/theora_image_transport/theora_publisher.h
index 6107c7f..5f151fc 100644
--- a/include/theora_image_transport/theora_publisher.h
+++ b/include/theora_image_transport/theora_publisher.h
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
diff --git a/include/theora_image_transport/theora_subscriber.h b/theora_image_transport/include/theora_image_transport/theora_subscriber.h
index 43dae13..e1907b8 100644
--- a/include/theora_image_transport/theora_subscriber.h
+++ b/include/theora_image_transport/theora_subscriber.h
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
diff --git a/package.xml b/theora_image_transport/package.xml
index e6b5e6b..4deb0a1 100644
--- a/package.xml
+++ b/package.xml
@@ -1,14 +1,15 @@
 <package>
   <name>theora_image_transport</name>
-  <version>1.14.0</version>
+  <version>1.15.0</version>
   <description>
     Theora_image_transport provides a plugin to image_transport for
     transparently sending an image stream encoded with the Theora codec.
   </description>
-  <maintainer email="jkammerl@willowgarage.com">Julius Kammerl</maintainer>
+  <maintainer email="kenjibrameld@gmail.com">Kenji Brameld</maintainer>
   <license>BSD</license>
 
   <url type="website">http://www.ros.org/wiki/image_transport_plugins</url>
+  <author>Julius Kammerl</author>
   <author>Patrick Mihelich</author>
   <author>Ethan Dreyfuss</author>
 
diff --git a/src/manifest.cpp b/theora_image_transport/src/manifest.cpp
index a00e8a6..1fce971 100644
--- a/src/manifest.cpp
+++ b/src/manifest.cpp
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
@@ -32,7 +32,7 @@
 *  POSSIBILITY OF SUCH DAMAGE.
 *********************************************************************/
 
-#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
 #include "theora_image_transport/theora_publisher.h"
 #include "theora_image_transport/theora_subscriber.h"
 
diff --git a/src/ogg_saver.cpp b/theora_image_transport/src/ogg_saver.cpp
index 5741aac..d59b62c 100644
--- a/src/ogg_saver.cpp
+++ b/src/ogg_saver.cpp
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
diff --git a/src/theora_publisher.cpp b/theora_image_transport/src/theora_publisher.cpp
index 2b2c7c4..4c4a93c 100644
--- a/src/theora_publisher.cpp
+++ b/src/theora_publisher.cpp
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
@@ -84,7 +84,7 @@ void TheoraPublisher::advertiseImpl(ros::NodeHandle &nh, const std::string &base
 
   // Set up reconfigure server for this topic
   reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
-  ReconfigureServer::CallbackType f = boost::bind(&TheoraPublisher::configCb, this, _1, _2);
+  ReconfigureServer::CallbackType f = boost::bind(&TheoraPublisher::configCb, this, boost::placeholders::_1, boost::placeholders::_2);
   reconfigure_server_->setCallback(f);
 }
 
diff --git a/src/theora_subscriber.cpp b/theora_image_transport/src/theora_subscriber.cpp
index 5997f7e..3f458b4 100644
--- a/src/theora_subscriber.cpp
+++ b/src/theora_subscriber.cpp
@@ -1,7 +1,7 @@
 /*********************************************************************
 * Software License Agreement (BSD License)
 * 
-*  Copyright (c) 20012, Willow Garage, Inc.
+*  Copyright (c) 2012, Willow Garage, Inc.
 *  All rights reserved.
 * 
 *  Redistribution and use in source and binary forms, with or without
@@ -73,7 +73,7 @@ void TheoraSubscriber::subscribeImpl(ros::NodeHandle &nh, const std::string &bas
 
   // Set up reconfigure server for this topic
   reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
-  ReconfigureServer::CallbackType f = boost::bind(&TheoraSubscriber::configCb, this, _1, _2);
+  ReconfigureServer::CallbackType f = boost::bind(&TheoraSubscriber::configCb, this, boost::placeholders::_1, boost::placeholders::_2);
   reconfigure_server_->setCallback(f);
 }
 
