diff --git a/setup.cfg b/setup.cfg
index 7568235..d3d6523 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -38,6 +38,7 @@ tests_require =
   flake8-import-order
   flake8-quotes
   pep8-naming
+  pydocstyle<6.0.0
   pylint
   pytest
   pytest-cov
@@ -47,6 +48,7 @@ zip_safe = false
 [tool:pytest]
 filterwarnings =
     error
+    ignore:lib2to3 package is deprecated::scspell
 junit_suite_name = colcon-bash
 
 [options.entry_points]
@@ -58,3 +60,4 @@ colcon_bash.shell.template = *.em
 
 [flake8]
 import-order-style = google
+ignore=E741
