diff --git a/pluginlib/include/pluginlib/class_loader_imp.hpp b/pluginlib/include/pluginlib/class_loader_imp.hpp
index 8ecd66e..db8a23e 100644
--- a/pluginlib/include/pluginlib/class_loader_imp.hpp
+++ b/pluginlib/include/pluginlib/class_loader_imp.hpp
@@ -85,7 +85,8 @@ NOINLINE ClassLoader<T>::ClassLoader(
   lowlevel_class_loader_(false)
   /***************************************************************************/
 {
-  ROS_DEBUG_NAMED("pluginlib.ClassLoader", "Creating ClassLoader, base = %s, address = %p",
+  ROS_DEBUG_STREAM_NAMED("pluginlib.ClassLoader",
+    "Creating ClassLoader, base = " << base_class.c_str() << ", address = " << this);
     base_class.c_str(), this);
   if (ros::package::getPath(package_).empty()) {
     throw pluginlib::ClassLoaderException("Unable to find package: " + package_);
@@ -95,9 +96,8 @@ NOINLINE ClassLoader<T>::ClassLoader(
     plugin_xml_paths_ = getPluginXmlPaths(package_, attrib_name_);
   }
   classes_available_ = determineAvailableClasses(plugin_xml_paths_);
-  ROS_DEBUG_NAMED("pluginlib.ClassLoader",
-    "Finished constructring ClassLoader, base = %s, address = %p",
-    base_class.c_str(), this);
+  ROS_DEBUG_STREAM_NAMED("pluginlib.ClassLoader",
+    "Finished constructring ClassLoader, base = " << base_class.c_str() << ", address = " << this);
 }
 
 template<class T>
