diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index fc60a02..a7d5129 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -67,22 +67,24 @@ if(WGET_EXE)
                    "--language=c++" "${PROJECT_SOURCE_DIR}/src" "${PROJECT_SOURCE_DIR}/include"
   )
 
-  add_custom_target(wget_cpplint
-                    COMMAND wget -q -O cpplint.py https://raw.githubusercontent.com/ament/ament_lint/master/ament_cpplint/ament_cpplint/cpplint.py
-  )
-
-  add_test(NAME console_bridge_cpplint
-           COMMAND "${PYTHON_EXECUTABLE}"
-                   "cpplint.py"
-                   "--counting=detailed"
-                   "--extensions=cpp,h"
-                   "--linelength=100"
-                   "--filter=-build/c++11,-runtime/references,-whitespace/braces,-whitespace/indent,-whitespace/parens,-whitespace/semicolon"
-           "${PROJECT_SOURCE_DIR}/include/console_bridge/console.h"
-           "${PROJECT_SOURCE_DIR}/src/console.cpp"
-  )
-
-  add_dependencies(console_bridge wget_cppcheck wget_cpplint)
+#  note: following lint command doesn't work correctly on aports build directory structure
+#
+#  add_custom_target(wget_cpplint
+#                    COMMAND wget -q -O cpplint.py https://raw.githubusercontent.com/ament/ament_lint/master/ament_cpplint/ament_cpplint/cpplint.py
+#  )
+#
+#  add_test(NAME console_bridge_cpplint
+#           COMMAND "${PYTHON_EXECUTABLE}"
+#                   "cpplint.py"
+#                   "--counting=detailed"
+#                   "--extensions=cpp,h"
+#                   "--linelength=100"
+#                   "--filter=-build/c++11,-runtime/references,-whitespace/braces,-whitespace/indent,-whitespace/parens,-whitespace/semicolon"
+#           "${PROJECT_SOURCE_DIR}/include/console_bridge/console.h"
+#           "${PROJECT_SOURCE_DIR}/src/console.cpp"
+#  )
+#
+#  add_dependencies(console_bridge wget_cppcheck wget_cpplint)
 
 else()
   message(WARNING "wget not found. Linters will not be executed")
