diff --git a/CMakeLists.txt b/CMakeLists.txt
index 415c236..c5bb5cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,6 +114,11 @@ if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   find_package(ament_cmake_gtest)
 
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+      ament_cmake_copyright
+      ament_cmake_cpplint
+      ament_cmake_uncrustify
+  )
   ament_lint_auto_find_test_dependencies()
 
   ament_add_gtest(${PROJECT_NAME}-point_cloud_common test/test_point_cloud_common.cpp)
