diff --git a/rosidl_cmake/__init__.py b/rosidl_cmake/__init__.py
index 0c7d1ed..333359b 100644
--- a/rosidl_cmake/__init__.py
+++ b/rosidl_cmake/__init__.py
@@ -93,7 +93,7 @@ def generate_files(
             print(
                 'Error processing idl file: ' +
                 str(locator.get_absolute_path()), file=sys.stderr)
-            raise(e)
+            raise (e)
 
     return generated_files
 
