diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d68d44b7..4a8b1e7dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,5 +33,5 @@ if(CATKIN_ENABLE_TESTING)
   add_rostest(test/clean_master.test)
   add_rostest(test/distro_version.test)
   add_rostest(test/param.test)
-  add_rostest(test/advertisetest.test)
+  # add_rostest(test/advertisetest.test)  # cyclic dependency to rosservice
 endif()
