diff --git a/CMake/FindCSparse.cmake b/CMake/FindCSparse.cmake
index 053c4e7..216374d 100644
--- a/CMake/FindCSparse.cmake
+++ b/CMake/FindCSparse.cmake
@@ -10,6 +10,8 @@ FIND_PATH(CSPARSE_INCLUDE_DIR NAMES cs.h
   /opt/local/include/ufsparse
   /usr/local/include/ufsparse
   /sw/include/ufsparse
+  PATH_SUFFIXES
+  suitesparse/suitesparse
 )
 
 FIND_LIBRARY(CSPARSE_LIBRARY NAMES cxsparse
@@ -18,8 +20,10 @@ FIND_LIBRARY(CSPARSE_LIBRARY NAMES cxsparse
   /usr/local/lib
   /opt/local/lib
   /sw/lib
+  PATH_SUFFIXES
+  suitesparse/suitesparse
 )
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(CSparse DEFAULT_MSG
-  CSPARSE_INCLUDE_DIR CSPARSE_LIBRARY)
\ No newline at end of file
+  CSPARSE_INCLUDE_DIR CSPARSE_LIBRARY)
