diff --git a/examples_rclpy_minimal_client/client_async_callback.py b/examples_rclpy_minimal_client/client_async_callback.py
index 7e8cd88..3b4ce6b 100644
--- a/examples_rclpy_minimal_client/client_async_callback.py
+++ b/examples_rclpy_minimal_client/client_async_callback.py
@@ -30,7 +30,7 @@ def main(args=None):
     did_get_result = False
 
     async def call_service():
-        nonlocal cli, node, did_run, did_get_result
+        nonlocal did_run, did_get_result
         did_run = True
         try:
             req = AddTwoInts.Request()
