diff --git a/src/lttngpy/channel.cpp b/src/lttngpy/channel.cpp
index 1402f31..e07792f 100644
--- a/src/lttngpy/channel.cpp
+++ b/src/lttngpy/channel.cpp
@@ -12,12 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Include lttng-error.h before channel.h, otherwise lttng_error_code isn't known
+// TODO(mosfet80): order it alphabetically once liblttng-ctl 2.14+ is fixed:
+// https://review.lttng.org/c/lttng-tools/+/15165
+#include <lttng/lttng-error.h>
 #include <lttng/channel.h>
 #include <lttng/constant.h>
 #include <lttng/domain.h>
 #include <lttng/event.h>
 #include <lttng/handle.h>
-#include <lttng/lttng-error.h>
 
 #include <optional>
 #include <string>
