diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f15ed8..f6bffb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,13 +126,11 @@ if(BUILD_TESTING)
   find_package(ament_cmake_cppcheck REQUIRED)
   find_package(ament_cmake_cpplint REQUIRED)
   find_package(ament_cmake_lint_cmake REQUIRED)
-  find_package(ament_cmake_uncrustify REQUIRED)
   find_package(rosgraph_msgs REQUIRED)
 
   ament_cppcheck(LANGUAGE "c++")
   ament_cpplint()
   ament_lint_cmake()
-  ament_uncrustify(LANGUAGE "c++")
 
   find_package(ament_cmake_gtest REQUIRED)
 
