diff --git a/include/octomap/OcTreeBase.h b/include/octomap/OcTreeBase.h
index d53b405..1e60049 100644
--- a/include/octomap/OcTreeBase.h
+++ b/include/octomap/OcTreeBase.h
@@ -43,7 +43,7 @@ namespace octomap {
   template <class NODE>
   class OcTreeBase : public OcTreeBaseImpl<NODE,AbstractOcTree> {
   public:
-    OcTreeBase<NODE>(double res) : OcTreeBaseImpl<NODE,AbstractOcTree>(res) {};
+    OcTreeBase(double res) : OcTreeBaseImpl<NODE,AbstractOcTree>(res) {}
 
     /// virtual constructor: creates a new object of same type
     /// (Covariant return type requires an up-to-date compiler)
