diff --git a/src/behavior_tree_engine.cpp b/src/behavior_tree_engine.cpp
index a9b43c7..fbcca45 100644
--- a/src/behavior_tree_engine.cpp
+++ b/src/behavior_tree_engine.cpp
@@ -76,6 +76,14 @@ BehaviorTreeEngine::run(
           1.0 / (loopRate.period().count() * 1.0e-9));
       }
     }
+  } catch (const BT::NodeExecutionError & ex) {
+    RCLCPP_ERROR(
+      rclcpp::get_logger("BehaviorTreeEngine"),
+      "BT Exception at Node: [%s] (Path: %s). Original error: %s. Exiting with failure.",
+      ex.failedNode().registration_name.c_str(),
+      ex.failedNode().node_path.c_str(),
+      ex.originalMessage().c_str());
+    return BtStatus::FAILED;
   } catch (const std::exception & ex) {
     RCLCPP_ERROR(
       rclcpp::get_logger("BehaviorTreeEngine"),
diff --git a/test/plugins/control/test_pipeline_sequence.cpp b/test/plugins/control/test_pipeline_sequence.cpp
index 517ee1f..72edfde 100644
--- a/test/plugins/control/test_pipeline_sequence.cpp
+++ b/test/plugins/control/test_pipeline_sequence.cpp
@@ -62,7 +62,7 @@ PipelineSequenceTestFixture::third_child_ = nullptr;
 TEST_F(PipelineSequenceTestFixture, test_failure_on_idle_child)
 {
   first_child_->changeStatus(BT::NodeStatus::IDLE);
-  EXPECT_THROW(bt_node_->executeTick(), std::runtime_error);
+  EXPECT_THROW(bt_node_->executeTick(), BT::NodeExecutionError);
 }
 
 TEST_F(PipelineSequenceTestFixture, test_failure)
diff --git a/test/plugins/control/test_recovery_node.cpp b/test/plugins/control/test_recovery_node.cpp
index 8292703..dfe3265 100644
--- a/test/plugins/control/test_recovery_node.cpp
+++ b/test/plugins/control/test_recovery_node.cpp
@@ -103,10 +103,10 @@ TEST_F(RecoveryNodeTestFixture, test_running)
 TEST_F(RecoveryNodeTestFixture, test_failure_on_idle_child)
 {
   first_child_->changeStatus(BT::NodeStatus::IDLE);
-  EXPECT_THROW(bt_node_->executeTick(), BT::LogicError);
+  EXPECT_THROW(bt_node_->executeTick(), BT::NodeExecutionError);
   first_child_->changeStatus(BT::NodeStatus::FAILURE);
   second_child_->changeStatus(BT::NodeStatus::IDLE);
-  EXPECT_THROW(bt_node_->executeTick(), BT::LogicError);
+  EXPECT_THROW(bt_node_->executeTick(), BT::NodeExecutionError);
 }
 
 TEST_F(RecoveryNodeTestFixture, test_success_one_retry)
diff --git a/test/plugins/control/test_round_robin_node.cpp b/test/plugins/control/test_round_robin_node.cpp
index 7daa91c..07ae0e3 100644
--- a/test/plugins/control/test_round_robin_node.cpp
+++ b/test/plugins/control/test_round_robin_node.cpp
@@ -58,7 +58,7 @@ std::shared_ptr<nav2_behavior_tree::DummyNode> RoundRobinNodeTestFixture::third_
 TEST_F(RoundRobinNodeTestFixture, test_failure_on_idle_child)
 {
   first_child_->changeStatus(BT::NodeStatus::IDLE);
-  EXPECT_THROW(bt_node_->executeTick(), BT::LogicError);
+  EXPECT_THROW(bt_node_->executeTick(), BT::NodeExecutionError);
 }
 
 TEST_F(RoundRobinNodeTestFixture, test_failure)
diff --git a/test/plugins/decorator/test_path_longer_on_approach.cpp b/test/plugins/decorator/test_path_longer_on_approach.cpp
index 5d521e2..1cc0f56 100644
--- a/test/plugins/decorator/test_path_longer_on_approach.cpp
+++ b/test/plugins/decorator/test_path_longer_on_approach.cpp
@@ -137,7 +137,7 @@ TEST_F(PathLongerOnApproachTestFixture, test_tick)
 
   // set new path on blackboard
   new_path.poses.resize(11);
-  for (unsigned int i = 0; i <= new_path.poses.size(); i++) {
+  for (unsigned int i = 0; i < new_path.poses.size(); i++) {
     // Assuming distance between waypoints to be 1.5m
     new_path.poses[i].pose.position.x = 1.5 * i;
   }
