diff --git a/CMakeLists.txt b/CMakeLists.txt
index d5a0bac..711195e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,9 @@ ament_target_dependencies(urdf_parser_plugin INTERFACE
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+  )
   set(ament_cmake_cppcheck_LANGUAGE "c++")
   set(ament_cmake_uncrustify_ADDITIONAL_ARGS LANGUAGE "C++")
   ament_lint_auto_find_test_dependencies()
