diff --git a/test/test_robot_utils.cpp b/test/test_robot_utils.cpp
index d8922868..fbc3d8ec 100644
--- a/test/test_robot_utils.cpp
+++ b/test/test_robot_utils.cpp
@@ -32,6 +32,7 @@ TEST(RobotUtils, LookupExceptionError)
   ASSERT_FALSE(nav2_util::getCurrentPose(global_pose, tf, "map", "base_link", 0.1));
   global_pose.header.frame_id = "base_link";
   ASSERT_FALSE(nav2_util::transformPoseInTargetFrame(global_pose, global_pose, tf, "map", 0.1));
+  rclcpp::shutdown();
 }
 
 TEST(RobotUtils, validateTwist)
