diff --git a/include/planner_cspace/planner_3d/grid_astar_model.h b/include/planner_cspace/planner_3d/grid_astar_model.h
index 63e217e..7d0f44b 100644
--- a/include/planner_cspace/planner_3d/grid_astar_model.h
+++ b/include/planner_cspace/planner_3d/grid_astar_model.h
@@ -30,6 +30,7 @@
 #ifndef PLANNER_CSPACE_PLANNER_3D_GRID_ASTAR_MODEL_H
 #define PLANNER_CSPACE_PLANNER_3D_GRID_ASTAR_MODEL_H
 
+#include <array>
 #include <memory>
 #include <utility>
 #include <vector>
diff --git a/test/src/test_blockmem_gridmap_performance.cpp b/test/src/test_blockmem_gridmap_performance.cpp
index 8cad9ca..63693b5 100644
--- a/test/src/test_blockmem_gridmap_performance.cpp
+++ b/test/src/test_blockmem_gridmap_performance.cpp
@@ -27,6 +27,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <array>
 #include <cstddef>
 #include <memory>
 
