diff --git a/ros2action/verb/send_goal.py b/ros2action/verb/send_goal.py
index 0ce2a28..73e4b92 100644
--- a/ros2action/verb/send_goal.py
+++ b/ros2action/verb/send_goal.py
@@ -145,7 +145,6 @@ def send_goal(action_name, action_type, goal_values, feedback_callback, timeout=
 
         # install signal handler to cancel the goal on SIGINT
         def _sigint_cancel_handler(sig, frame):
-            nonlocal goal_handle
             # Cancel the goal if it's still active
             if (goal_handle is not None and
                 (GoalStatus.STATUS_ACCEPTED == goal_handle.status or
