diff --git a/launch_testing/launch_testing/markers.py b/launch_testing/launch_testing/markers.py
index bb15cab..d070967 100644
--- a/launch_testing/launch_testing/markers.py
+++ b/launch_testing/launch_testing/markers.py
@@ -49,7 +49,11 @@ def retry_on_failure(*, times, delay=None):
                         assert self._outcome.success
                     return ret
                 except AssertionError:
-                    self._outcome.errors.clear()
+                    if hasattr(self._outcome, 'errors'):
+                        self._outcome.errors.clear()
+                    else:
+                        if self._outcome.result is not None:
+                            self._outcome.result.errors.clear()
                     self._outcome.success = True
                     n -= 1
                 if delay is not None:
