diff --git a/CMakeLists.txt b/CMakeLists.txt
index 36e4972..c0c87d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,12 +95,10 @@ if(BUILD_TESTING)
   find_package(ament_cmake_cppcheck REQUIRED)
   find_package(ament_cmake_cpplint REQUIRED)
   find_package(ament_cmake_lint_cmake REQUIRED)
-  find_package(ament_cmake_uncrustify REQUIRED)
   #TODO(mjcarroll) Headers need to be .hpp for this to work properly.
   #ament_cppcheck()
   ament_cpplint()
   ament_lint_cmake()
-  ament_uncrustify()
   find_package(ament_cmake_gtest)
 
   ament_add_gtest(${PROJECT_NAME}-parse_ini test/test_parse_ini.cpp)
