diff --git a/common/include/pcl/conversions.h b/common/include/pcl/conversions.h
index 20072819d..896d8abb1 100644
--- a/common/include/pcl/conversions.h
+++ b/common/include/pcl/conversions.h
@@ -176,7 +176,6 @@ namespace pcl
     // check if there is data to copy
     if (msg.width * msg.height == 0)
     {
-      PCL_WARN("[pcl::fromPCLPointCloud2] No data to copy.\n");
       return;
     }
 
