diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0632bc..3e76b9d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,9 @@ macro(build_mimick)
     GIT_CONFIG advice.detachedHead=false
     # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
     UPDATE_COMMAND ""
+    PATCH_COMMAND
+        COMMAND sed -i "s|ElfAddr|Elf64_Addr|g" src/plt-elf.c
+        COMMAND sed -i "s|ElfOff|Elf64_Off|g" src/plt-elf.c
     TIMEOUT 6000
     ${cmake_commands}
     CMAKE_ARGS
