diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2926c71..c9b7e1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,10 @@ find_package(rmw_implementation_cmake REQUIRED)
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   ament_lint_auto_find_test_dependencies()
 endif()
 
