diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9685445..e13e62b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,9 @@ if(BUILD_TESTING)
   find_package(rosidl_cmake REQUIRED)
 
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+  )
   ament_lint_auto_find_test_dependencies()
 
   find_package(test_interface_files REQUIRED)
