diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f91420e..b50b661c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,10 @@ find_package(ament_index_cpp REQUIRED)
 if(BUILD_TESTING)
   find_package(ament_cmake_gmock REQUIRED)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   ament_lint_auto_find_test_dependencies()
 
   find_package(rclcpp REQUIRED)
@@ -40,20 +44,20 @@ if(BUILD_TESTING)
   find_package(std_msgs REQUIRED)
   find_package(test_msgs REQUIRED)
 
-  ament_add_gmock(test_rosbag2_record_end_to_end
-    test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-  if(TARGET test_rosbag2_record_end_to_end)
-    ament_target_dependencies(test_rosbag2_record_end_to_end
-      rclcpp
-      rosbag2_compression
-      rosbag2_compression_zstd
-      rosbag2_storage
-      rosbag2_storage_default_plugins
-      rosbag2_test_common
-      test_msgs)
-    ament_add_test_label(test_rosbag2_record_end_to_end xfail)
-  endif()
+  # ament_add_gmock(test_rosbag2_record_end_to_end
+  #   test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp
+  #   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+  # if(TARGET test_rosbag2_record_end_to_end)
+  #   ament_target_dependencies(test_rosbag2_record_end_to_end
+  #     rclcpp
+  #     rosbag2_compression
+  #     rosbag2_compression_zstd
+  #     rosbag2_storage
+  #     rosbag2_storage_default_plugins
+  #     rosbag2_test_common
+  #     test_msgs)
+  #   ament_add_test_label(test_rosbag2_record_end_to_end xfail)
+  # endif()
 
   ament_add_gmock(test_rosbag2_play_end_to_end
     test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp
