diff --git a/src/rosdistro/distribution_file.py b/src/rosdistro/distribution_file.py
index c330329..710978f 100644
--- a/src/rosdistro/distribution_file.py
+++ b/src/rosdistro/distribution_file.py
@@ -74,6 +74,9 @@ class DistributionFile(object):
                     assert os_code_name not in self.release_platforms[os_name], "Distribution '%s' specifies the os_code_name '%s' multiple times for the os_name '%s'" % (self.name, os_code_name, os_name)
                     self.release_platforms[os_name].append(os_code_name)
 
+        # Tweak for alpine platform
+        self.release_platforms['alpine'] = ['3.20']
+
         self.tags = []
         if 'tags' in data and data['tags']:
             for tag in data['tags']:
