diff --git a/tf2/src/cache.cpp b/tf2/src/cache.cpp
index f9614c7..3994ab2 100644
--- a/tf2/src/cache.cpp
+++ b/tf2/src/cache.cpp
@@ -40,6 +40,8 @@
 
 namespace tf2 {
 
+const static bool ALLOW_TF_REAPEATED = std::getenv("ALLOW_TF_REPEATED") == "true";
+
 TransformStorage::TransformStorage()
 {
 }
@@ -273,17 +275,24 @@ bool TimeCache::insertData(const TransformStorage& new_data, std::string* error_
       break;
     storage_it++;
   }
-  if (storage_it != storage_.end() && storage_it->stamp_ == new_data.stamp_)
+  if (ALLOW_TF_REAPEATED)
   {
-    if (error_str)
-    {
-      *error_str = "TF_REPEATED_DATA ignoring data with redundant timestamp";
-    }
-    return false;
+    storage_.insert(storage_it, new_data);
   }
   else
   {
-    storage_.insert(storage_it, new_data);
+    if (storage_it != storage_.end() && storage_it->stamp_ == new_data.stamp_)
+    {
+      if (error_str)
+      {
+        *error_str = "TF_REPEATED_DATA ignoring data with redundant timestamp";
+      }
+      return false;
+    }
+    else
+    {
+      storage_.insert(storage_it, new_data);
+    }
   }
 
   pruneList();
