diff --git a/setup.cfg b/setup.cfg
index 1497714..c7fd9c1 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -54,6 +54,7 @@ share/colcon_argcomplete/hook =
 [tool:pytest]
 filterwarnings =
     error
+    ignore:lib2to3 package is deprecated::scspell
 junit_suite_name = colcon-argcomplete
 
 [options.entry_points]
@@ -66,3 +67,4 @@ colcon_core.extension_point =
 
 [flake8]
 import-order-style = google
+ignore=E741
