diff --git a/include/usb_cam/formats/mjpeg.hpp b/include/usb_cam/formats/mjpeg.hpp
index 2faff23..2b40bb2 100644
--- a/include/usb_cam/formats/mjpeg.hpp
+++ b/include/usb_cam/formats/mjpeg.hpp
@@ -233,7 +233,7 @@ private:
     std::cerr << m_averror_str << std::endl;
   }
 
-  AVCodec * m_avcodec;
+  const AVCodec * m_avcodec;
   AVCodecContext * m_avcodec_context;
   AVCodecParserContext * m_avparser;
   AVFrame * m_avframe_device;
