diff --git a/CMakeLists.txt b/CMakeLists.txt
index 88bf75d2..92291c7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,11 +95,11 @@ if(BUILD_TESTING)
   add_definitions(-D_TEST_RESOURCES_DIR_PATH="${CMAKE_CURRENT_SOURCE_DIR}/test/${PROJECT_NAME}")
 
   set(ament_cmake_clang_format_CONFIG_FILE .clang-format)
-  list(APPEND AMENT_LINT_AUTO_EXCLUDE ament_cmake_uncrustify)
-  if(WIN32)
-    # clang-format is not easily exposed to Windows, this linter enforced regardless via Linux/OSX builds
-    list(APPEND AMENT_LINT_AUTO_EXCLUDE ament_cmake_clang_format)
-  endif()
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_uncrustify
+    ament_cmake_copyright
+    ament_cmake_clang_format
+  )
   ament_lint_auto_find_test_dependencies()
 
   ament_add_gmock(test_mcap_storage test/rosbag2_storage_mcap/test_mcap_storage.cpp)
