diff --git a/test/test_timer.py b/test/test_timer.py
index 394f750..95015f1 100644
--- a/test/test_timer.py
+++ b/test/test_timer.py
@@ -138,7 +138,7 @@ def test_cancel_reset(period):
                     timer.reset()
                     assert not timer.is_canceled()
                     begin_time = time.time()
-                    while rclpy.ok(context=context) and time.time() - begin_time < 2.5 * period:
+                    while rclpy.ok(context=context) and time.time() - begin_time < 2.9 * period:
                         executor.spin_once(timeout_sec=period / 10)
                     assert len(callbacks) == 2
                 finally:
