diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ddb208..d16bd82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_python REQUIRED)
+find_package(ament_cmake_ros REQUIRED)
 find_package(lifecycle_msgs REQUIRED)
 find_package(rcl REQUIRED)
 find_package(rcl_action REQUIRED)
@@ -138,7 +139,12 @@ if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   # Give cppcheck hints about macro definitions coming from outside this package
   get_target_property(ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS rcutils::rcutils INTERFACE_INCLUDE_DIRECTORIES)
-  list(APPEND AMENT_LINT_AUTO_EXCLUDE "ament_cmake_cppcheck")
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    "ament_cmake_cppcheck"
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+    ament_cmake_flake8
+  )
   ament_lint_auto_find_test_dependencies()
 
   find_package(ament_cmake_cppcheck REQUIRED)
@@ -219,7 +225,7 @@ if(BUILD_TESTING)
 
     foreach(_test_path ${_rclpy_pytest_tests})
       get_filename_component(_test_name ${_test_path} NAME_WE)
-      ament_add_pytest_test(${_test_name} ${_test_path}
+      ament_add_ros_isolated_pytest_test(${_test_name} ${_test_path}
         APPEND_ENV AMENT_PREFIX_PATH=${ament_index_build_path}
           PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}
         TIMEOUT 120
