diff --git a/src/camera_nodelet.cpp b/src/camera_nodelet.cpp
index 1c1a7d8..b320295 100644
--- a/src/camera_nodelet.cpp
+++ b/src/camera_nodelet.cpp
@@ -24,6 +24,8 @@
 #include <string>
 #include <vector>
 
+#include <boost/shared_ptr.hpp>
+
 #include <cv_bridge/cv_bridge.h>
 #include <image_transport/image_transport.h>
 #include <nodelet/nodelet.h>
@@ -468,7 +470,7 @@ ifm3d_ros::CameraNodelet::Run()
       ros::Duration(1.0).sleep();
     }
 
-  pcl::PointCloud<ifm3d::PointT>::Ptr
+  boost::shared_ptr<pcl::PointCloud<ifm3d::PointT>>
     cloud(new pcl::PointCloud<ifm3d::PointT>());
 
   cv::Mat confidence_img;
