diff --git a/test/test_model_parser_initxml.cpp b/test/test_model_parser_initxml.cpp
index 6acf9d8..7b33ae4 100644
--- a/test/test_model_parser_initxml.cpp
+++ b/test/test_model_parser_initxml.cpp
@@ -58,7 +58,8 @@ TEST(model_parser_initxml, initstring_good)
 TEST(model_parser_initxml, initxml_tinyxml_element_bad)
 {
   urdf::Model model;
-  ASSERT_FALSE(model.initXml(reinterpret_cast<TiXmlElement *>(NULL)));
+  TiXmlElement *p = nullptr;
+  ASSERT_FALSE(model.initXml(p));
 }
 
 TEST(model_parser_initxml, initxml_tinyxml_element_good)
@@ -73,7 +74,8 @@ TEST(model_parser_initxml, initxml_tinyxml_element_good)
 TEST(model_parser_initxml, initxml_tinyxml_document_bad)
 {
   urdf::Model model;
-  ASSERT_FALSE(model.initXml(reinterpret_cast<TiXmlDocument *>(NULL)));
+  TiXmlDocument *p;
+  ASSERT_FALSE(model.initXml(p));
 }
 
 TEST(model_parser_initxml, initxml_tinyxml_document_good)
@@ -88,7 +90,8 @@ TEST(model_parser_initxml, initxml_tinyxml_document_good)
 TEST(model_parser_initxml, initxml_tinyxml2_element_bad)
 {
   urdf::Model model;
-  ASSERT_FALSE(model.initXml(reinterpret_cast<tinyxml2::XMLElement *>(NULL)));
+  tinyxml2::XMLElement *p = nullptr;
+  ASSERT_FALSE(model.initXml(p));
 }
 
 TEST(model_parser_initxml, initxml_tinyxml2_element_good)
@@ -103,7 +106,8 @@ TEST(model_parser_initxml, initxml_tinyxml2_element_good)
 TEST(model_parser_initxml, initxml_tinyxml2_document_bad)
 {
   urdf::Model model;
-  ASSERT_FALSE(model.initXml(reinterpret_cast<tinyxml2::XMLDocument *>(NULL)));
+  tinyxml2::XMLDocument *p = nullptr;
+  ASSERT_FALSE(model.initXml(p));
 }
 
 TEST(model_parser_initxml, initxml_tinyxml2_document_good)
