diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e8d654..40ddc57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,7 +46,6 @@ install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
 
 # Tests
 if(BUILD_TESTING)
-  find_package(ament_cmake_copyright REQUIRED)
   find_package(ament_cmake_cppcheck REQUIRED)
   find_package(ament_cmake_cpplint REQUIRED)
   find_package(ament_cmake_lint_cmake REQUIRED)
@@ -65,7 +64,6 @@ if(BUILD_TESTING)
     include/tf2/LinearMath/Vector3.h
   )
 
-  ament_copyright(EXCLUDE ${_linter_excludes})
   ament_cppcheck(
     EXCLUDE ${_linter_excludes}
     LANGUAGE c++
