diff --git a/CMakeLists.txt b/CMakeLists.txt
index d42ce68..83f789c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,6 +67,9 @@ ament_index_register_resource("rosidl_typesupport_cpp")
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+  )
   find_package(osrf_testing_tools_cpp REQUIRED)
 
   find_package(performance_test_fixture REQUIRED)
