diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c4a5f4..173b9ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,10 @@ ament_export_dependencies("tlsf")
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   ament_lint_auto_find_test_dependencies()
 
   # get typesupport of rmw implementation to include / link against the corresponding interfaces
