diff --git a/cmake/cpptoml/CMakeLists.txt b/cmake/cpptoml/CMakeLists.txt
index abcdefg..1234567 100644
--- a/cmake/cpptoml/CMakeLists.txt
+++ b/cmake/cpptoml/CMakeLists.txt
@@ -72,7 +72,7 @@ if(DEFINED CMAKE_TOOLCHAIN_FILE)
     set(cpptoml_DIR ${cpptoml_DIR} CACHE PATH "" FORCE)
 endif()

-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCPPTOML_BUILD_EXAMPLES=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${SOURCE_DIR}" ${CMAKE_ADDITIONAL_OPTIONS}
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCPPTOML_BUILD_EXAMPLES=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "-DCMAKE_POLICY_VERSION_MINIMUM=3.5" "${SOURCE_DIR}" ${CMAKE_ADDITIONAL_OPTIONS}
     RESULT_VARIABLE result
     WORKING_DIRECTORY ${BUILD_DIR} )
 if(result)
