diff --git a/test/test_ament_mypy.py b/test/test_ament_mypy.py
index a47730e..0398077 100644
--- a/test/test_ament_mypy.py
+++ b/test/test_ament_mypy.py
@@ -260,4 +260,4 @@ def test__get_xunit_content(mocker, sample_errors):
                                              [err.group('filename') for err in errors], .01)
     root = ET.fromstring(xml)
     assert root.get('failures') == '0'
-    assert(len(root)) == 2
+    assert (len(root)) == 2
