diff --git a/CMakeLists.txt b/CMakeLists.txt
index df200309..aba9ff32 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -141,6 +141,9 @@ install(DIRECTORY plugins/
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+  )
   find_package(ament_cmake_gtest REQUIRED)
   add_subdirectory(test)
 endif()
