diff --git a/src/libros/connection_manager.cpp b/src/libros/connection_manager.cpp
index c2a98e9d5..512af199e 100644
--- a/src/libros/connection_manager.cpp
+++ b/src/libros/connection_manager.cpp
@@ -178,7 +178,7 @@ void ConnectionManager::udprosIncomingConnection(const TransportUDPPtr& transpor
   ConnectionPtr conn(boost::make_shared<Connection>());
   addConnection(conn);
 
-  conn->initialize(transport, true, NULL);
+  conn->initialize(transport, true, HeaderReceivedFunc());
   onConnectionHeaderReceived(conn, header);
 }
 
diff --git a/src/libros/subscription.cpp b/src/libros/subscription.cpp
index 81cd3449b..21cc1a955 100644
--- a/src/libros/subscription.cpp
+++ b/src/libros/subscription.cpp
@@ -583,7 +583,7 @@ void Subscription::pendingConnectionDone(const PendingConnectionPtr& conn, XmlRp
     if (pub_link->setHeader(h))
     {
       ConnectionPtr connection(boost::make_shared<Connection>());
-      connection->initialize(udp_transport, false, NULL);
+      connection->initialize(udp_transport, false, HeaderReceivedFunc());
       connection->setHeader(h);
       pub_link->initialize(connection);
 
