diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19715083..281b0d09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,10 @@ install(DIRECTORY include/
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  list(APPEND AMENT_LINT_AUTO_EXCLUDE
+    ament_cmake_copyright
+    ament_cmake_uncrustify
+  )
   find_package(ament_cmake_gtest REQUIRED)
   ament_lint_auto_find_test_dependencies()
   add_subdirectory(test)
