diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 77870826..0ed4aa75 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,10 +26,10 @@ target_link_libraries(polygons_test
 )
 
 # Collision Monitor node test
-ament_add_gtest(collision_monitor_node_test collision_monitor_node_test.cpp)
-ament_target_dependencies(collision_monitor_node_test
-  ${dependencies}
-)
-target_link_libraries(collision_monitor_node_test
-  ${library_name}
-)
+# ament_add_gtest(collision_monitor_node_test collision_monitor_node_test.cpp)
+# ament_target_dependencies(collision_monitor_node_test
+#   ${dependencies}
+# )
+# target_link_libraries(collision_monitor_node_test
+#   ${library_name}
+# )
