diff --git a/src/imu_filter_ros.cpp b/src/imu_filter_ros.cpp
index ed8bd98..c9766aa 100644
--- a/src/imu_filter_ros.cpp
+++ b/src/imu_filter_ros.cpp
@@ -137,12 +137,14 @@ ImuFilterRos::ImuFilterRos(ros::NodeHandle nh, ros::NodeHandle nh_private)
     int queue_size = 5;
 
     imu_subscriber_.reset(new ImuSubscriber(
-        nh_, ros::names::resolve("imu") + "/data_raw", queue_size));
+        nh_, ros::names::resolve("imu") + "/data_raw", queue_size,
+        ros::TransportHints().tcpNoDelay(true)));
 
     if (use_mag_)
     {
         mag_subscriber_.reset(new MagSubscriber(
-            nh_, ros::names::resolve("imu") + "/mag", queue_size));
+            nh_, ros::names::resolve("imu") + "/mag", queue_size,
+            ros::TransportHints().tcpNoDelay(true)));
 
         sync_.reset(new Synchronizer(SyncPolicy(queue_size), *imu_subscriber_,
                                      *mag_subscriber_));
