diff --git a/ament_clang_tidy/main.py b/ament_clang_tidy/main.py
index 6745d8d..37d72fb 100755
--- a/ament_clang_tidy/main.py
+++ b/ament_clang_tidy/main.py
@@ -154,7 +154,7 @@ def main(argv=sys.argv[1:]):
             cmd.append('--system-headers')

         def is_gtest_source(file_name):
-            if(file_name == 'gtest_main.cc' or file_name == 'gtest-all.cc'
+            if (file_name == 'gtest_main.cc' or file_name == 'gtest-all.cc'
                or file_name == 'gmock_main.cc' or file_name == 'gmock-all.cc'):
                 return True
             return False

