diff --git a/osrf_pycommon/process_utils/execute_process_nopty.py b/osrf_pycommon/process_utils/execute_process_nopty.py
index fd22aee..78214f2 100644
--- a/osrf_pycommon/process_utils/execute_process_nopty.py
+++ b/osrf_pycommon/process_utils/execute_process_nopty.py
@@ -111,7 +111,7 @@ def _yield_data(p, fds, left_overs, linesep, fds_to_close=None):
                     # this error does not occur repeatedly (it does not
                     # become a busy wait). See:
                     #   http://stackoverflow.com/a/12207447/671658
-                    if _is_linux and "Input/output error" in "{0}".format(exc):
+                    if _is_linux and "[Errno 5]" in "{0}".format(exc):
                         continue
                     raise
                 if not incoming:
