diff --git a/test/map_saver_cli/test_map_saver_cli.cpp b/test/map_saver_cli/test_map_saver_cli.cpp
index 79cf5899..cd46e140 100644
--- a/test/map_saver_cli/test_map_saver_cli.cpp
+++ b/test/map_saver_cli/test_map_saver_cli.cpp
@@ -143,4 +143,6 @@ TEST(MapSaverCLI, CLITest)
     "ros2 run nav2_map_server map_saver_cli --ros-args -r __node:=map_saver_test_node");
   return_code = system(command.c_str());
   EXPECT_EQ(return_code, 0);
+
+  rclcpp::shutdown();
 }
